internal/pkg/agent/application/fleet_server_bootstrap.go

```diff
@@ -82,6 +83,37 @@ func FleetServerComponentModifier(serverCfg *configuration.FleetServerConfig) co
 	}
 }
 
+func EndpointComponentModifier(fleetCfg *configuration.FleetAgentConfig) coordinator.ComponentsModifier {
```

Comment:
Can you document what this does, and why it is needed (i.e Endpoint currently needs to talk directly to Fleet server to upload files)?

Reply:
Added comments